### PR TITLE
feat: introduce status indicator component

### PR DESCRIPTION
**Pull request by Lokanathan-k**
introduce status indicator component

## Related Issue
https://github.com/SAP/fundamental-styles/issues/2021

## Description
 Introduce status indicator component styles 

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: stories/status-indicator/status-indicator.stories.js [hunk 1]
```diff
@@ -0,0 +1,637 @@
...
+					c-0.9-0.7-1.5-1.7-2-3H6.4l0.5-1.9h1V12H6.4l0.5-1.9h1.3C8.6,8.9,9.3,8,10.3,7.3c1-0.8,2.3-1.1,3.8-1.1c1.1,0,2,0.2,2.8,0.7L17.3,7
+					l-0.6,2.3l-0.5-0.4c-0.4-0.2-0.9-0.4-1.4-0.5c-0.6-0.1-1.1-0.1-1.7,0c-0.6,0.1-1,0.3-1.5,0.7c-0.3,0.3-0.6,0.6-0.7,1H16.4z" stroke-width="0.25" stroke="#6a6d70" fill="transparent">
+				</path>
+			</svg>
+		</svg>
+	</svg>
+</div>
+</div>
+<div class="example-container">
+<span style="min-width: 150px;">Critical Filling :</span>
```
Comment (kavya-b) [inline stories/status-indicator/status-indicator.stories.js:206]: inline styling is not encouraged. Can you place documentation related custom styling in custom.scss?

File: src/status-indicator.scss [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  color: $fd-status-indicator-default-color;
+  stroke: $fd-status-indicator-default-color-stroke;
+  stroke-width: $fd-status-indicator-default-color-stroke-width;
+  fill: $fd-status-indicator-default-color-fill;
+  flex-direction: column;
+  align-items: center;
+
+  @include fd-reset();
+
+  &-htext {
```
Comment (kavya-b) [inline src/status-indicator.scss:38]: this should probably be a modifier

**Comment by kavya-b** [comment]
all examples not showing correct dotted focus: 
image
image

same titles for examples? 
image

quartz dark doesn't work right in IE, could you check?
image

**Comment by sKudum** [comment]
Hi lok,

**Query**.
 Are we targeting to provide svgs to users? as per my understanding application's UX team will create it and user has to tag them in html code and apply our modifier classes.

**Few observations.**
1).Having examples for Cirulcar fill, filling sequence, grouping, thresholds will help users.
2).How about radial and angular mechanisms, could not find any.
3).Sharing few details about usage will help, for example how can achieve required stroke, which class will help me to achieve radial mechanism, how I need to use it.
4). How I have to use these classes when I have a svg image shared from ux team?
           Try to use sketch for having one svg stored. in local, point out how a svg file and apply the classes for desired mechanisum.

_As mentioned before, we can take help of fiori-moments. how those guys are doing it._

**Comment by sKudum** [comment]
Few Quires:

1). On click , on hover, active etc these svgs shouldn't have a11y features?
2).Are we targeting to give classes which help users to apply desired color and size in svg.  Everthing else its has to take help from svg?
3).Is it possible to have documentation with more details.
            we can state which options are from lib and from svg to obtain desired behaviour.?

**Comment by Lokanathan-k** [comment]
> On click , on hover, active etc these svgs shouldn't have a11y features?

@puru-hk i have added the a11y feature as part of the pr, pls review this if it needs more details on the a11y for the above component.

**Comment by Lokanathan-k** [comment]
> Few Quires:
> 
> 1). On click , on hover, active etc these svgs shouldn't have a11y features?
these on hover or active etc are not defined in the spec.
> 2).Are we targeting to give classes which help users to apply desired color and size in svg. Everthing else its has to take help from svg?
yes, we can only achieve the size and color other filling is done by lineargradient and radialgradient.
> 3).Is it possible to have documentation with more details.
added
> we can state which options are from lib and from svg to obtain desired behaviour.?

**Review by DeepakSap14 — APPROVED** [review]

**Review by sKudum — APPROVED** [review]
Approving it to avoid blocking, expecting queries be addressed with respected person consultation.

**Comment by puru-hk** [comment]
Tested: From accessibility point of view able access view keyboard, tooltip is implemented for icon. 
But regarding the screen reader it is not reading the image icon.

**Review by olshansky — APPROVED** [review]

**Review by JKMarkowski — CHANGES_REQUESTED** [review]
Hello @Lokanathan-k 
Could you add commit with `[ci visual]` to trigger chromatic? After it I can approve - Good Job!

**Comment by puru-hk** [comment]
Looks fine for me

**Review by puru-hk — APPROVED** [review]

**Review by JKMarkowski — APPROVED** [review]

**Pull request merged**